tests/ssl-opt.sh

```diff
@@ -8007,6 +8007,69 @@ run_test    "ECJPAKE: working, TLS" \
             -S "SSL - The handshake negotiation failed" \
             -S "SSL - Verification of the message MAC failed"
 
+requires_config_enabled MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED
+requires_config_enabled MBEDTLS_USE_PSA_CRYPTO
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
+run_test    "ECJPAKE_OPAQUE_PW: working, TLS, opaque password client+server" \
+            "$P_SRV debug_level=3 ecjpake_pw=bla ecjpake_pw_opaque=1" \
+            "$P_CLI debug_level=3 ecjpake_pw=bla ecjpake_pw_opaque=1\
+             force_ciphersuite=TLS-ECJPAKE-WITH-AES-128-CCM-8" \
+            0 \
+            -c "add ciphersuite: c0ff" \
+            -c "adding ecjpake_kkpp extension" \
+            -c "using opaque password" \
+            -s "using opaque password" \
+            -C "re-using cached ecjpake parameters" \
+            -s "found ecjpake kkpp extension" \
+            -S "skip ecjpake kkpp extension" \
+            -S "ciphersuite mismatch: ecjpake not configured" \
+            -s "server hello, ecjpake kkpp extension" \
+            -c "found ecjpake_kkpp extension" \
+            -S "SSL - The handshake negotiation failed" \
+            -S "SSL - Verification of the message MAC failed"
+
+requires_config_enabled MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED
+requires_config_enabled MBEDTLS_USE_PSA_CRYPTO
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
+run_test    "ECJPAKE_OPAQUE_PW: working, TLS, opaque password client only" \
```

Comment:
If there's no need for a shared `ECJPAKE_OPAQUE_PW` begining of the test name (as a smaller subset of tests should be run in the `all.sh` component), this can be changed to `ECJPAKE: working [...]`

Reply:
Actually I think I need this `ECJPAKE_OPAQUE_PW` label in order not to execute some tests when the `component_test_tls1_2_ecjpake_compatibility` runs.

Currently tests in `ssl-opt.sh` should be OK as long as both client and server run with or without USE_PSA. If you try to run `ECJPAKE_OPAQUE_PW` tests when USE_PSA is disabled, then those tests are skipped by the `requires_config_enabled MBEDTLS_USE_PSA_CRYPTO`.

If a mixed condition is used, then it's not possible to tell from the current configuration file which of the two has USE_PSA enabled. I mean, in theory you can based on the order you used to build executables, but it doesn't look very elegant to me.

That's why I kept it for now. However if this is a relevant problem, I can try to find an alternative solution.

Reply:
Let's continue the discussion in the comment above :)